src/main/java/com/igot/cb/discussion/service/impl/DiscussionServiceImpl.java

```diff
@@ -997,4 +1038,122 @@ public ApiResponse updateAnswerPost(JsonNode answerPostData, String token) {
+            response.setResult(map);
+            return response;
+        } catch (Exception e) {
+            log.error("DiscussionService::bookmarkDiscussion: Failed to bookmark discussion", e);
+            return returnErrorMsg(Constants.DISCUSSION_BOOKMARK_FAILED, HttpStatus.INTERNAL_SERVER_ERROR, response, Constants.FAILED);
+        }
+    }
+
+    public ApiResponse unBookmarkDiscussion(String communityId, String discussionId, String token) {
+        log.info("DiscussionService::unBookmarkDiscussion: UnBookmarking discussion");
+        ApiResponse response = ProjectUtil.createDefaultResponse("discussion.unBookmarkDiscussion");
+        if (StringUtils.isBlank(discussionId)) {
+            return returnErrorMsg(Constants.INVALID_DISCUSSION_ID, HttpStatus.BAD_REQUEST, response, Constants.FAILED);
+        }
+
+        if (StringUtils.isBlank(communityId)) {
+            return returnErrorMsg(Constants.INVALID_DISCUSSION_ID, HttpStatus.BAD_REQUEST, response, Constants.FAILED);
+        }
+        String userId = accessTokenValidator.verifyUserToken(token);
+        if (StringUtils.isBlank(userId) || Constants.UNAUTHORIZED.equals(userId)) {
+            return returnErrorMsg(Constants.INVALID_AUTH_TOKEN, HttpStatus.UNAUTHORIZED, response, Constants.FAILED);
+        }
+
+        Map<String, Object> bookmarkData = new HashMap<>();
+        bookmarkData.put(Constants.COMMUNITY_ID, communityId);
+        bookmarkData.put(Constants.DISCUSSION_ID, discussionId);
+        bookmarkData.put(Constants.USERID, userId);
+        try {
+            cassandraOperation.deleteRecord(Constants.KEYSPACE_SUNBIRD, Constants.DISCUSSION_BOOKMARKS, bookmarkData);
+            return response;
+        } catch (Exception e) {
+            log.error("DiscussionService::unBookmarkDiscussion: Failed to unBookmark discussion", e);
+            return returnErrorMsg(Constants.DISCUSSION_UN_BOOKMARK_FAILED, HttpStatus.INTERNAL_SERVER_ERROR, response, Constants.FAILED);
+        }
+    }
+
+    @Override
+    public ApiResponse getBookmarkedDiscussions(String token) {
+        log.info("DiscussionService::getBookmarkedDiscussions: Fetching bookmarked discussions");
+        ApiResponse response = ProjectUtil.createDefaultResponse("discussion.getBookmarkedDiscussions");
+
+        String userId = accessTokenValidator.verifyUserToken(token);
+        if (StringUtils.isBlank(userId) || Constants.UNAUTHORIZED.equals(userId)) {
+            return returnErrorMsg(Constants.INVALID_AUTH_TOKEN, HttpStatus.UNAUTHORIZED, response, Constants.FAILED);
+        }
+
+        try {
+            Map<String, Object> properties = new HashMap<>();
+            properties.put(Constants.USERID, userId);
+            List<Map<String, Object>> bookmarkedDiscussions = cassandraOperation.getRecordsByPropertiesWithoutFiltering(
+                    Constants.KEYSPACE_SUNBIRD, Constants.DISCUSSION_BOOKMARKS, properties, Arrays.asList(Constants.DISCUSSION_ID), null);
+
```

Comment:
limit should be applied. results should be grouped on community id ?
do check with UI team how they need the response - how to sort ?